### PR TITLE
Added a super snowflake vendor for devs and admins to play with

### PR DESCRIPTION
**Pull request by ThePiachu**
## About The Pull Request

I was doing some work on picking a set of clothes for a fun ERT when I realised we could use a vendor for admins / devs to figure out what kind of items look good together and so on before they spawn a bunch of mobs with them. So I added a few Super Snowflake Vendors that have all the items of a given type in them. The UI is a bit slow due to the number of items, and when I tried putting all of the items together into one vendor it lagged out the game, so I split them up into separate types.

With some feedback from carlarc, I also made the Super Snowflake vendor be easily customizable - if you spawn the base variant it will ask you to provide an item path to stock. You can also easily configure it while mapping to stock a given set of subitems.

## Why It's Good For The Game

It makes it easier to match interesting outfit combinations for ERTs and what have you. This can let admins communicate more clearly what they might want in a gear preset, and for devs to easier test how things look together.

## Changelog

:cl:
add: Added a set of Super Snowflake Vendors for admins and devs to play with. If you spawn the base one you can stock it with any item and its subtypes as defined when you first spawn it / interact with it.
/:cl:

### CODE REVIEW

**Comment by fira** [comment]
I like the idea already but lemme put forward one even more crazy:
What if instead this was a blank vendor, and the adding functionality was a tool/verb rather than thru usage? So you can use it multiple times to make a custom vendor.
Might even be extended to other vendors, though i don't think most would support this properly...

**Comment by ThePiachu** [comment]
Okay, so I changed the vendor like this:

1) It takes a list of objects in now, so it would be easy to dynamically make a vendor on the map or something
2) Admins got a verb to add items to the vendor, so they can create an empty vendor and populate it with stuff they want to give to people

This should be enough to make a custom vendor on the fly and have it dispense the needed items.

Not sure how we're handling such admin verbs, should it be in this file or somewhere else...

**Comment by ThePiachu** [comment]
So @fira - is this how you wanted the vendor to function, or should I use some kind of different setup for admin adding new things to it?

**Review by harryob — APPROVED** [review]

File: code/game/machinery/vending/vendor_types/dress.dm [hunk 1]
```diff
@@ -160,3 +160,100 @@
...
+	//If we don't have an object type, we ask the user to supply it
+	if(!item_types)
+		var/obj/item/chosen = get_item_category_from_mob(user)
+		if(!chosen)
+			return
+		item_types = list(chosen)
+		to_world("item_types - [item_types.len]")
+
+	if(!items)
+		to_world("item_types - [item_types.len]")
```
Comment (harryob) [inline code/game/machinery/vending/vendor_types/dress.dm:187]: debug?

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/game/machinery/vending/vendor_types/dress.dm (hunk 1 above)
Comment (ThePiachu) [reply code/game/machinery/vending/vendor_types/dress.dm:187]: debug...

**Review by harryob — APPROVED** [review]

**Review by harryob — CHANGES_REQUESTED** [review]
seems like every item is out of stock - regardless of if you use a preset vendor or add to one yourself?

**Comment by ThePiachu** [comment]
Ah, my mistake, had to make it "use points" while still keeping everything at 0 points. Should be good now.

**Review by harryob — DISMISSED** [review]
apologies - missed this the first time around: but if a type is selected with subtypes, every subtype is also added, and it's not possible to just add the specific type suggested, as far as i can tell. is this intentional?

**Review by harryob — APPROVED** [review]

**Pull request merged**